suite/tinytables/src/main/java/dk/alexandra/fresco/suite/tinytables/prepro/TinyTablesPreproResourcePool.java

```diff
@@ -2,62 +2,104 @@
+import dk.alexandra.fresco.suite.tinytables.ot.TinyTablesOt;
 import dk.alexandra.fresco.suite.tinytables.prepro.protocols.TinyTablesPreproANDProtocol;
 import dk.alexandra.fresco.suite.tinytables.prepro.protocols.TinyTablesPreproProtocol;
 import dk.alexandra.fresco.suite.tinytables.storage.BatchTinyTablesTripleProvider;
 import dk.alexandra.fresco.suite.tinytables.storage.TinyTablesStorage;
 import dk.alexandra.fresco.suite.tinytables.storage.TinyTablesStorageImpl;
 import dk.alexandra.fresco.suite.tinytables.storage.TinyTablesTripleProvider;
 import dk.alexandra.fresco.suite.tinytables.util.Util;
+import dk.alexandra.fresco.tools.cointossing.CoinTossing;
+import dk.alexandra.fresco.tools.ot.base.Ot;
+import dk.alexandra.fresco.tools.ot.otextension.BristolOtFactory;
+import dk.alexandra.fresco.tools.ot.otextension.OtExtensionResourcePool;
+import dk.alexandra.fresco.tools.ot.otextension.OtExtensionResourcePoolImpl;
+import dk.alexandra.fresco.tools.ot.otextension.RotFactory;
+import dk.alexandra.fresco.tools.ot.otextension.RotList;
+
 import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.io.ObjectOutputStream;
 import java.nio.ByteBuffer;
-import java.security.SecureRandom;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.Comparator;
 import java.util.List;
-import java.util.Random;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class TinyTablesPreproResourcePool extends ResourcePoolImpl {
 
-  private static final Logger logger = LoggerFactory.getLogger(TinyTablesPreproResourcePool.class);
+  private static final Logger LOGGER = LoggerFactory.getLogger(TinyTablesPreproResourcePool.class);
+  private static final int OT_BATCH_SIZE = 16000;
 
-  private final Random random;
+  private final Drng random;
   private final List<TinyTablesPreproANDProtocol> unprocessedAnds;
   private final TinyTablesStorage storage;
   private final File tinyTablesFile;
+  private final TinyTablesOt baseOt;
+  private final RotList rotList;
+  private final CoinTossing ct;
+  private final OtExtensionResourcePool otExtRes;
   private TinyTablesTripleProvider tinyTablesTripleProvider;
 
   /**
    * Creates an instance of the default implementation of a resource pool. This contains the basic
    * resources needed within FRESCO.
    *
    * @param myId The ID of the MPC party.
+   * @param baseOt OT functionality for the base OTs
+   * @param random Secure randomness generator
    * @param noOfPlayers The amount of parties within the MPC computation.
    * @param tinyTablesFile file for data
    */
-  public TinyTablesPreproResourcePool(int myId, int noOfPlayers,
-      File tinyTablesFile) {
+  public TinyTablesPreproResourcePool(int myId, int noOfPlayers, TinyTablesOt baseOt, Drbg random,
+      int computationalSecurity, int statisticalSecurity, File tinyTablesFile) {
     super(myId, noOfPlayers);
-    this.random = new SecureRandom();
+    if (noOfPlayers != 2) {
```

Comment:
Why then even take this parameter? Does it have any use other than throwing an exception here?

Reply:
Removed.